string transition
string marking
int64 robot_id
string timestamp
